Makefile

```diff
@@ -0,0 +1,50 @@
+FEATURES_MINIMAL = --no-default-features --features 'std cargo' -p clap:3.0.0-beta.5
```

Comment:
> Previously, the flag -p clap:3.0.0-beta.4 was removed from the command which is supposed to lint our repo when almost no features are enabled. This would not work at all because the other packages (which are now run) enable the default features for clap. So, I added it back. (Tested this thoroughly)

Like with https://github.com/clap-rs/clap/pull/2903, have we audited those dependencies to see if any should be removed?

Reply:
`clap_derive` definitely needs `derive` dependency which we can't remove. So, we can't achieve minimal testing without providing `-p clap:3.0.0-beta.5`

Reply:
Been giving this some thought over the weekend.

Especially as feature flags add/remove parts of the API (e.g. `env`), we need to test the whole monorepo against `minimal` and not just `clap`.

Some ideas
- We leave off `-p`, assuming testing with `derive` is of a minimal risk from the loss of coverage of testing without it since it has little API or behavior impact (only if a plugin derived rather than hand-implemented a trait)
- Move `-p` to the CI pipeline and have the job do a run without it and then a run with it (should have minimal overhead since it'll mostly be able to reuse build results)
- Move the derive tests out of `clap_derive` and put a `#![cfg(feature = "derive")]` in then (since we treat `clap_derive` as more of an implementation detail of `clap`).

Also in thinking about this more, this is mostly independent of this PR (this PR just adds it back in one place and moves where the flag exists), so feel free to defer this if you want.

(Before I forget, I also suggest we switch to `--manifest-path` from `-p` so we don't to reduce number of places updated on version, bump, manual or automated, keeping the git history more focused on what is relevant for someone digging into it)